lib/ex_json_schema/validator/format.ex

```diff
@@ -57,10 +54,25 @@ defmodule ExJsonSchema.Validator.Format do
     []
   end
 
+  defp do_validate(_, "date" = format, data) do
+    case Date.from_iso8601(data) do
+      {:ok, %Date{}} -> []
+      _ -> [%Error{error: %Error.Format{expected: format}}]
+    end
+  end
+
+  defp do_validate(_, "date-time" = format, data) do
+    data
+    |> String.upcase()
```

Comment:
The `String.upcase/1` call is needed since there's a test on the `date-time.json` test suite that has a lower `t` and `z`: https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/689d2f28182720d3778ca91c149f725a6ce73f71/tests/draft7/optional/format/date-time.json#L41-L45

Reply:
Seems like an acceptable work-around 👍